### PR TITLE
Implement get and list methods for tasks in OPI

### PR DESCRIPTION
**Pull request by gdankov**
Implement `fetch_task` and `fetch_tasks` methods in the OPI task client.

Fixes https://github.com/cloudfoundry/cf-for-k8s/issues/412.
Also, related: https://github.com/cloudfoundry-incubator/eirini/issues/107.

[#174335347]

Co-authored-by: Giuseppe Capizzi <gcapizzi@pivotal.io>

* [x] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/master/CONTRIBUTING.md)

* [x] I have viewed, signed, and submitted the Contributor License Agreement

* [x] I have made this pull request to the `master` branch

* [x] I have run all the unit tests using `bundle exec rake`

* [x] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

### CODE REVIEW

**Review by cwlbraa — COMMENTED** [review]
This looks straightforward and solid to me. I forwarded to the rest of the team for more eyes. https://cloudfoundry.slack.com/archives/C017LDM6KTQ/p1603139810196500

**Review by tcdowney — APPROVED** [review]
Looks good to me as well.

File: lib/cloud_controller/opi/task_client.rb [hunk 1]
```diff
@@ -21,12 +21,33 @@ def desire_task(task, domain)
       end
     end
 
-    def fetch_task(_)
-      Diego::Bbs::Models::Task.new
+    def fetch_task(guid)
+      resp = client.get("/tasks/#{guid}")
+
+      return nil if resp.status_code == 404
```
Comment (matt-royal) [inline lib/cloud_controller/opi/task_client.rb:27]: The "cannonical" Ruby form for `return nil` is `return`

**Comment by matt-royal** [comment]
This looks good. You may want to change `return nil` to `return` to make it "canonical" Ruby, but that's my only quibble.

**Comment by gdankov** [comment]
@matt-royal, fixed it.

**Pull request merged**